tests/cover/test_testdecorators.py

```diff
@@ -339,19 +339,17 @@ def test_has_ascii(x):
     assert any(c in ascii_characters for c in x)
 
 
-def test_uses_provided_seed():
-    import random
-    random.seed(0)
-    initial = random.getstate()
-
+def test_with_seed():
```

Comment:
@Zac-HD Here's a story about this test:

With the new PYTHONWARNINGS, this test still passes the 1st time, but then if it's run again it fails when we have examples for it in the database already:

```
hypothesis.errors.HypothesisDeprecationWarning: In future use of @seed will imply database=None in your settings, but your test is currently using examples from the database. To get the future behaviour, update your settings for this test to include database=None.
```

Then I thought: Okay, but isn't it weird that before we turned on errors-for-warnings this test was okay? If it's really testing that we use the seed, shouldn't it fail the 2nd time even before the PYTHONWARNINGS change?

Turns out the answer to that is: It's not really checking that the seed is used! The test still passes even with `@seed` removed, or if I replace the test with:

```
def test_uses_provided_seed():
    import random
    random.seed(0)
    initial = random.getstate()

    def seed(some_seed):
        def wrapper(f):
            print(some_seed)
            return f
        return wrapper

    @given(integers())
    @seed(42)
    @settings(database=None)
    def test_foo(x):
        pass

    test_foo()
    assert hash(repr(random.getstate())) == hash(repr(initial))
```

So here's my suggestion:

1. These changes do a few nice things: 
a. avoid failing on the 2nd run (when we see the warning and that turns into an error after this PR)
b. relatedly, ensure that the code we run actually tries to use the seed consistently even on the 2nd run
c. avoid pretending that we're checking whether the seed is used when we aren't

2. If you're okay with this plan / this PR, then I'll open an issue about the fact that we should change the test to check that the seed is actually used, linking back to this change & this comment for context.

I haven't had a chance to think about how the test was intended to work and why it isn't working as intended. If someone hands me a quick fix, I'm happy to put it in this PR, but otherwise I'd prefer for this PR not to be blocked on figuring that out, and these changes seem like good ones for now.

Reply:
Oh, correction, getting an error the 2nd time isn't actually new with this PR (we already error'd on such warnings), so I guess all that stuff is unrelated to this.

Reply:
... I'm going to put this in a separate CR since it's unrelated.

Reply:
and also I think I had totally misunderstood that test. Oh well / sorry, at least this thread is hidden behind my push...